File: test/engine/golden/layered-postgresql-fr/src/main/java/mg/solumada/payflow/common/ConflictException.java
package mg.solumada.payflow.common;

/**
 * Conflit avec l'état courant (doublon, version obsolète) : traduit en 409 par ApiExceptionHandler.
 */
public class ConflictException extends RuntimeException {

    public ConflictException(String message) {
        super(message);
    }
}

File: test/engine/golden/aspnet-layered-postgresql-fr/src/Solumada.PayFlow.Domain/Common/ConflictException.cs
namespace Solumada.PayFlow.Domain.Common;

// CORE-011 : levée quand l'état demandé contredit l'état courant, traduite en 409.
public sealed class ConflictException : Exception
{
    public ConflictException()
        : base("Conflicting state") { }

    public ConflictException(string message)
        : base(message) { }

    public ConflictException(string message, Exception innerException)
        : base(message, innerException) { }
}
